src/inferencerequest.cpp

```diff
@@ -104,10 +104,14 @@ Status InferenceRequest::getBatchSize(size_t& batchSize, size_t batchSizeIndex)
     batchSize = shape[batchSizeIndex];
     return StatusCode::OK;
 }
+
+// Assuming the request is already validated, therefore no need to check for negative values or zeros
 std::map<std::string, shape_t> InferenceRequest::getRequestShapes() const {
     std::map<std::string, shape_t> result;
     for (auto& [name, tensor] : inputs) {
-        result.emplace(name, tensor.getShape());
+        result.emplace(name, shape_t(
+                                 tensor.getShape().data(),
```

Comment:
This is InferenceTensor which does have signed_shape_t type. Does this work?

Reply:
This works due to implicit cast from int64_t* to uint64_t*.
Changed to use `reinterpret_cast` to make it explicit.